File: dev-docs/snippets/collection-type-configs/create.swift
import JsBaoClient

// Create a new collection type configuration. Takes a typed
// `CreateCollectionTypeConfigParams`; `collectionType` is required, `ruleSetId`
// is optional.
func create(client: JsBaoClient, ruleSetId: String) async throws {
  // #region example
  let config = try await client.collectionTypeConfigs.create(
    params: CreateCollectionTypeConfigParams(
      collectionType: "class-students",
      ruleSetId: ruleSetId
    )
  )
  // #endregion example
  _ = config
}

File: dev-docs/snippets/collection-type-configs/create.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Create a new collection type configuration. `collectionType` is required;
// `ruleSetId` is optional.
export async function create(client: JsBaoClient, ruleSetId: string) {
  // #region example
  const config = await client.collectionTypeConfigs.create({
    collectionType: "class-students",
    ruleSetId,
  });
  // #endregion example
  return config;
}
